### DISCUSSION TITLE
Publish the DFC business vocabulary as a separated library

### DISCUSSION THREAD

**Discussion by lecoqlibre** [datafoodconsortium/connector-codegen#19] [Ideas]
Recently I published [@datafoodconsortium/vocab-business](https://www.npmjs.com/package/@datafoodconsortium/vocab-business) on NPM.

This TypeScript library contains the URL of all the predicates and classes of the DFC's ontology like `Address`, `affiliatedTo`, etc.

This way, we can replace the hard coded (generated) URL of each predicate and classe by its corresponding element of the library. For instance, instead of writing:
```ts
const type: string = "https://github.com/datafoodconsortium/ontology/releases/latest/download/DFC_BusinessOntology.owl#Catalog";
```

We could write:

```ts
import { DFC } from '@datafoodconsortium/vocab-business';

const type: string = DFC.Catalog; // or even better, remove this "type" variable and replace all occurrences with DFC.Catalog.
```

This offers some advantages: 
- decoupling the connector from the vocabulary so users would be able to use the vocabulary alone, 
- it is reducing the size of the connector,
- and we could update the vocabulary without updating the connector.

This would require to maintain one more library but like the other generators we should set up GitHub actions to deploy it automatically.

I wrote a dedicated generator, see PR https://github.com/datafoodconsortium/connector-codegen/pull/18.

I propose we do the same for all programming languages.

Any thoughts @RaggedStaff @mkllnk?

**Comment by mkllnk** [comment]
Hm, I wonder if we achieve the same by using only prefixes within the connector and the context is doing this mapping. You've been doing that in the latest ruby connector already.

I'm not keen on adding another dependency. But providing the vocabulary on its own may have a use for others. For us, the current connector solves the problem of mapping code to DFC types and we don't need the additional library.
